Migrate deprecated `recommonmark` to `myst-parser` (#1489)
kedro-org/kedro: docs/source/tools_integration/ipython.md

In [3]: %reload_kedro
```

```{note}

> If you want to pass arguments to the :code:`reload_kedro` line magic, e.g. to specify a configuration environment, you should call it as a normal Python function (:code:`reload_kedro(env=env, extra_params=extra_params`) rather than using :code:`%reload_kedro` in a Notebook cell (:code:`%reload_kedro(extra_params=extra_params)` wouldn’t work). You might have to call :code:`%automagic False` beforehand to make this work.

Author: ```suggestion
If you want to pass arguments to the `reload_kedro` line magic, e.g. to specify a configuration environment, you should call it as a normal Python function (`reload_kedro(env=env, extra_params=extra_params)`) rather than using `%reload_kedro` in a Notebook cell (`%reload_kedro(extra_params=extra_params)` wouldn't work). You might have to call `%automagic False` beforehand to make this work.
```

 'example_learning_rate': 0.01}
```

> ```{note}

Author: From documentation contributor guidelines: "Prefer to use `note`, `warning` and `important` only, rather than a number of different colours/types of callout." I tried to color match to the appropriate level.

> If you enable `versioning <../data/data_catalog.md#versioning-datasets-and-ml-models>`_ then you can load a particular version of a dataset, e.g. :code:`catalog.load("example_train_x", version="2021-12-13T15.08.09.255Z")`.

Author: ```suggestion
If you enable [versioning](../data/data_catalog.md#versioning-datasets-and-ml-models) then you can load a particular version of a dataset, e.g. `catalog.load("example_train_x", version="2021-12-13T15.08.09.255Z")`.
```

Out[1]: {}
```

```{note}

> You can only execute one *successful* run per session, as there's a one-to-one mapping between a session and a run. If you wish to do multiple runs, you'll have to run :code:`%reload_kedro` to obtain a new :code:`session`.

Author: ```suggestion
You can only execute one *successful* run per session, as there's a one-to-one mapping between a session and a run. If you wish to do multiple runs, you'll have to run `%reload_kedro` to obtain a new `session`.
```

We recommend that you store your Jupyter Notebooks in the `notebooks` folder of your Kedro project. If you are using `kedro jupyter notebook` or `kedro jupyter lab` then you should use the default kernel selected for you, which is listed as `Kedro (<project_package_name>)`. This will run the Kedro IPython extension automatically when the kernel is started, so that the `catalog`, `context`, `pipelines` and `session` variables are available immediately to you.

```{note}

> Restarting the kernel will reload the Kedro IPython extension and hence refresh the :code:`catalog`, :code:`context`, :code:`pipelines` and :code:`session` variables.

Author: ```suggestion
Restarting the kernel will reload the Kedro IPython extension and hence refresh the `catalog`, `context`, `pipelines` and `session` variables.
```

* Add the `node` tag to the cell containing your function
![Add the node tag graphic](../meta/images/jupyter_notebook_workflow_tagging_nodes.png)

> ```{note}

Author: From documentation contributor guidelines: "Prefer to use `note`, `warning` and `important` only, rather than a number of different colours/types of callout." I tried to color match to the appropriate level.

> The Notebook can contain multiple functions tagged as ``node``, each of them will be exported into the resulting Python file

Author: ```suggestion
The Notebook can contain multiple functions tagged as `node`, each of them will be exported into the resulting Python file
```

```

* Save your Jupyter Notebook to `notebooks/my_notebook.ipynb`
* Run `kedro jupyter convert notebooks/my_notebook.ipynb` from the terminal to create a Python file `src/<package_name>/nodes/my_notebook.py` containing `some_action` function definition

> ```{note}

Author: From documentation contributor guidelines: "Prefer to use `note`, `warning` and `important` only, rather than a number of different colours/types of callout." I tried to color match to the appropriate level.

> You can also convert all your Notebooks at once by calling ``kedro jupyter convert --all``.

Author: ```suggestion
You can also convert all your Notebooks at once by calling `kedro jupyter convert --all`.
```
